Http 500 while import semgrep results

**Bug description**
There is an error while import semgrep result using API or UI import function
Stripped semgrep output attached
[semgrep2.json.txt](https://github.com/DefectDojo/django-DefectDojo/files/12186553/semgrep2.json.txt)

**Steps to reproduce**
Steps to reproduce the behavior:
1. Take attached json and import in some engagement using Semgrep importer
2. See error "Well... ....this was unexpected"

**Expected behavior**
Semgrep output should be uploaded correctly

**Deployment method** *(select with an `X`)*
- [X ] Docker Compose

**Environment information**
 - Operating System: Debian 11
 - DefectDojo version (see footer) or commit message: 2.22.2

**Logs** 
```
django-defectdojo-uwsgi-1  | [27/Jul/2023 16:44:16] INFO [django.request:241] OK: /ddojo/engagement/60/import_scan_results
django-defectdojo-uwsgi-1  | Traceback (most recent call last):
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56, in inner
django-defectdojo-uwsgi-1  |     response = get_response(request)
django-defectdojo-uwsgi-1  |                ^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/deprecation.py", line 138, in __call__
django-defectdojo-uwsgi-1  |     response = self.process_response(request, response)
django-defectdojo-uwsgi-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/watson/middleware.py", line 37, in process_response
django-defectdojo-uwsgi-1  |     self._close_search_context(request)
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/watson/middleware.py", line 33, in _close_search_context
django-defectdojo-uwsgi-1  |     search_context_manager.end()
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 261, in end
django-defectdojo-uwsgi-1  |     list(chain.from_iterable(engine._update_obj_index_iter(obj)
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 505, in _update_obj_index_iter
django-defectdojo-uwsgi-1  |     "content": adapter.get_content(obj),
django-defectdojo-uwsgi-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 147, in get_content
django-defectdojo-uwsgi-1  |     return self.prepare_content(" ".join(
django-defectdojo-uwsgi-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 101, in prepare_content
django-defectdojo-uwsgi-1  |     content = strip_tags(content)
django-defectdojo-uwsgi-1  |               ^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/functional.py", line 246, in wrapper
django-defectdojo-uwsgi-1  |     return func(*args, **kwargs)
django-defectdojo-uwsgi-1  |            ^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/html.py", line 175, in strip_tags
django-defectdojo-uwsgi-1  |     new_value = _strip_once(value)
django-defectdojo-uwsgi-1  |                 ^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/html.py", line 163, in _strip_once
django-defectdojo-uwsgi-1  |     s.feed(value)
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/html/parser.py", line 110, in feed
django-defectdojo-uwsgi-1  |     self.goahead(0)
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/html/parser.py", line 178, in goahead
django-defectdojo-uwsgi-1  |     k = self.parse_html_declaration(i)
django-defectdojo-uwsgi-1  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/html/parser.py", line 263, in parse_html_declaration
django-defectdojo-uwsgi-1  |     return self.parse_marked_section(i)
django-defectdojo-uwsgi-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-defectdojo-uwsgi-1  |   File "/usr/local/lib/python3.11/_markupbase.py", line 154, in parse_marked_section
django-defectdojo-uwsgi-1  |     raise AssertionError(
django-defectdojo-uwsgi-1  | AssertionError: unknown status keyword 'A-Z' in marked section

```
**Sample scan files**
attached

**Screenshots**
nope

**Additional context** (optional)
I think this error is from bad html-tag-like content in field "lines" of json file
This content comes from code snippet, which semgrep extracted from defective code

We're seeing the same when importing SARIF results (since it flows down into the same culprit code) on DefectDojo v2.25.2:
```
[01/Sep/2023 05:00:08] INFO [django.request:241] OK: /api/v2/import-scan/
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/deprecation.py", line 138, in __call__
    response = self.process_response(request, response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/watson/middleware.py", line 37, in process_response
    self._close_search_context(request)
  File "/usr/local/lib/python3.11/site-packages/watson/middleware.py", line 33, in _close_search_context
    search_context_manager.end()
  File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 261, in end
    list(chain.from_iterable(engine._update_obj_index_iter(obj)
  File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 505, in _update_obj_index_iter
    "content": adapter.get_content(obj),
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 147, in get_content
    return self.prepare_content(" ".join(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/watson/search.py", line 101, in prepare_content
    content = strip_tags(content)
              ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/functional.py", line 246, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/html.py", line 175, in strip_tags
    new_value = _strip_once(value)
                ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/html.py", line 163, in _strip_once
    s.feed(value)
  File "/usr/local/lib/python3.11/html/parser.py", line 110, in feed
    self.goahead(0)
  File "/usr/local/lib/python3.11/html/parser.py", line 178, in goahead
    k = self.parse_html_declaration(i)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/html/parser.py", line 263, in parse_html_declaration
    return self.parse_marked_section(i)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_markupbase.py", line 154, in parse_marked_section
    raise AssertionError(
AssertionError: unknown status keyword 'A-Z' in marked section
```

--- a/dojo/tools/semgrep/parser.py
+++ b/dojo/tools/semgrep/parser.py
@@ -101,6 +101,10 @@ class SemgrepParser(object):
 
         snippet = item["extra"].get("lines")
         if snippet is not None:
-            description += "**Snippet:**\n```{}```\n".format(snippet)
+            if "<![" in snippet:
+                snippet = snippet.replace("<![", "<! [")
+                description += "**Snippet:** ***Caution:*** Please remove the space between `!` and `[` to have the real value due to a workaround to circumvent [#8435](https://github.com/DefectDojo/django-DefectDojo/issues/8435).\n```{}```\n".format(snippet)
+            else:
+                description += "**Snippet:**\n```{}```\n".format(snippet)
 
         return description

--- a/unittests/scans/semgrep/issue_8435.json
+++ b/unittests/scans/semgrep/issue_8435.json
@@ -1,0 +1,69 @@
+{
+    "errors": [
+    ],
+    "results": [
+        {
+            "check_id": "generic.secrets.security.detected-twitter-oauth.detected-twitter-oauth",
+            "end": {
+                "col": 38431,
+                "line": 1,
+                "offset": 38430
+            },
+            "extra": {
+                "engine_kind": "OSS",
+                "fingerprint": "fc255e575f2b938f11bf70bb737def0a1aa3ea044296972c9c6ff5b66b0d931064731b30406fc873b300c2bcf73cb66797513b709991e334fd355599d60d53d8_0",
+                "is_ignored": false,
+                "lines": "<![A-Z]>",
+                "message": "Twitter OAuth detected",
+                "metadata": {
+                    "category": "security",
+                    "confidence": "LOW",
+                    "cwe": [
+                        "CWE-798: Use of Hard-coded Credentials"
+                    ],
+                    "cwe2021-top25": true,
+                    "cwe2022-top25": true,
+                    "impact": "MEDIUM",
+                    "license": "Commons Clause License Condition v1.0[LGPL-2.1-only]",
+                    "likelihood": "LOW",
+                    "owasp": [
+                        "A07:2021 - Identification and Authentication Failures"
+                    ],
+                    "references": [
+                        "https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures"
+                    ],
+                    "semgrep.dev": {
+                        "rule": {
+                            "origin": "community",
+                            "rule_id": "BYUNq8",
+                            "url": "https://semgrep.dev/playground/r/A8TRG6/generic.secrets.security.detected-twitter-oauth.detected-twitter-oauth",
+                            "version_id": "A8TRG6"
+                        }
+                    },
+                    "shortlink": "https://sg.run/Lwb7",
+                    "source": "https://semgrep.dev/r/generic.secrets.security.detected-twitter-oauth.detected-twitter-oauth",
+                    "source-rule-url": "https://github.com/dxa4481/truffleHogRegexes/blob/master/truffleHogRegexes/regexes.json",
+                    "subcategory": [
+                        "audit"
+                    ],
+                    "technology": [
+                        "secrets",
+                        "twitter"
+                    ],
+                    "vulnerability_class": [
+                        "Hard-coded Secrets"
+                    ]
+                },
+                "metavars": {},
+                "severity": "ERROR"
+            },
+            "path": "/somedir/somefile.js",
+            "start": {
+                "col": 37650,
+                "line": 1,
+                "offset": 37649
+            }
+        }
+  ],
+    "version": "1.33.2"
+}

--- a/unittests/tools/test_semgrep_parser.py
+++ b/unittests/tools/test_semgrep_parser.py
@@ -121,3 +121,10 @@ class TestSemgrepParser(DojoTestCase):
         self.assertEqual(len(findings_first), len(findings_second))
         for first, second in zip(findings_first, findings_second):
             self.assertEqual(first.unique_id_from_tool, second.unique_id_from_tool)
+
+    def test_parse_issue_8435(self):
+        testfile = open("unittests/scans/semgrep/issue_8435.json")
+        parser = SemgrepParser()
+        findings = parser.get_findings(testfile, Test())
+        testfile.close()
+        self.assertEqual(1, len(findings))

